src/SeleniumLibrary/keywords/element.py

```diff
@@ -112,21 +113,42 @@ def page_should_contain_element(self, locator, message=None, loglevel='INFO'):
         The ``message`` argument can be used to override the default error
         message.
 
+        The ``limit`` argument can used to define how many elements the
+        page should contain. When ``limit`` is `None` page can contain one or
+        more elements. When limit is a number, page must contain same number
+        of elements.
+
         See `Page Should Contain` for explanation about the ``loglevel``
         argument.
+
+        The ``limit`` argument is new in SeleniumLibrary 3.0
+
+        Examples assumes that locator matches to two elements.
+        | Page Should Contain Element | name: div_name | limit=1    | # Keyword fails.                  |
+        | Page Should Contain Element | name: div_name | limit=2    | # Keyword passes.                 |
+        | Page Should Contain Element | name: div_name | limit=none | # None is considered one or more. |
+        | Page Should Contain Element | name: div_name |            | # Same as above.                  |
         """
-        self.assert_page_contains(locator, message=message, loglevel=loglevel)
+        if is_noney(limit):
+            self.assert_page_contains(locator, message=message,
+                                      loglevel=loglevel)
+        else:
```

Comment:
This nested if/else structure is pretty complex. I'd add `return` at the end of `if` here and remove `else` altogether. Or perhaps the whole `else` content should be in a separate helper to follow the [single level of abstraction principle](http://www.markhneedham.com/blog/2009/06/12/coding-single-level-of-abstraction-principle/).

Reply:
Having the else is waste. I was also thinking creating a helper, but then decided against it.